src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -636,6 +643,52 @@ def post_remove(self, daemon: DaemonDescription, is_failed_deploy: bool) -> None
         # super().post_remove(daemon)
         pass
 
+    def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
+        daemon_spec.final_config, daemon_spec.deps = super().generate_config(daemon_spec)
+
+        mon_spec = cast(MONSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
+        if mon_spec.crush_locations:
+            if daemon_spec.host in mon_spec.crush_locations:
+                # the --crush-location flag only supports a single bucker=loc pair so
+                # others will have to be handled later. The idea is to set the flag
+                # for the first bucket=loc pair in the list in order to facilitate
+                # replacing a tiebreaker mon (https://docs.ceph.com/en/quincy/rados/operations/stretch-mode/#other-commands)
+                c_loc = mon_spec.crush_locations[daemon_spec.host][0]
+                daemon_spec.final_config['crush_location'] = c_loc
+
+        return daemon_spec.final_config, daemon_spec.deps
+
+    def set_crush_locations(self, daemon_descrs: List[DaemonDescription], spec: ServiceSpec) -> None:
+        if not daemon_descrs:
+            return
+        assert self.TYPE == spec.service_type
+        mon_spec = cast(MONSpec, spec)
+
+        if mon_spec.crush_locations:
+            quorum_status = self._get_quorum_status()
```

Comment:
nic: probably we can just check  `if  not mon_spec.crush_locations: return` to a large indentation on the whole function body. WDYT?